firebase-firestore/src/test/java/com/google/firebase/firestore/remote/BloomFilterTest.java

```diff
@@ -0,0 +1,252 @@
+
+  @Test
+  public void instantiateEmptyBloomFilter() {
+    BloomFilter bloomFilter = new BloomFilter(new byte[0], 0, 0);
+    assertEquals(bloomFilter.getBitCount(), 0);
+  }
+
+  @Test
+  public void instantiateNonEmptyBloomFilter() {
+    BloomFilter bloomFilter1 = new BloomFilter(new byte[] {1}, 0, 1);
+    assertEquals(bloomFilter1.getBitCount(), 8);
+    BloomFilter bloomFilter2 = new BloomFilter(new byte[] {1}, 7, 1);
+    assertEquals(bloomFilter2.getBitCount(), 1);
+  }
+
+  @Test
+  public void constructorShouldThrowNPEOnNullBitmap() {
+    NullPointerException emptyBloomFilterException =
+        assertThrows(NullPointerException.class, () -> new BloomFilter(null, 0, 0));
+    assertThat(emptyBloomFilterException).hasMessageThat().contains("Bitmap cannot be null.");
+    NullPointerException nonEmptyBloomFilterException =
+        assertThrows(NullPointerException.class, () -> new BloomFilter(null, 1, 1));
+    assertThat(nonEmptyBloomFilterException).hasMessageThat().contains("Bitmap cannot be null.");
+  }
+
+  @Test
+  public void constructorShouldThrowIAEOnEmptyBloomFilterWithNonZeroPadding() {
+    IllegalArgumentException exception =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[0], 1, 0));
+    assertThat(exception).hasMessageThat().contains("Invalid padding when bitmap length is 0: 1");
+  }
+
+  @Test
+  public void constructorShouldThrowIAEOnNonEmptyBloomFilterWithZeroHashCount() {
+    IllegalArgumentException zeroHashCountException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[] {1}, 1, 0));
+    assertThat(zeroHashCountException).hasMessageThat().contains("Invalid hash count: 0");
+  }
+
+  @Test
+  public void constructorShouldThrowIAEOnNegativePadding() {
+    IllegalArgumentException emptyBloomFilterException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[0], -1, 0));
+    assertThat(emptyBloomFilterException).hasMessageThat().contains("Invalid padding: -1");
+    IllegalArgumentException nonEmptyBloomFilterException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[] {1}, -1, 1));
+    assertThat(nonEmptyBloomFilterException).hasMessageThat().contains("Invalid padding: -1");
+  }
+
+  @Test
+  public void constructorShouldThrowIAEOnNegativeHashCount() {
+    IllegalArgumentException emptyBloomFilterException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[0], 0, -1));
+    assertThat(emptyBloomFilterException).hasMessageThat().contains("Invalid hash count: -1");
+    IllegalArgumentException nonEmptyBloomFilterException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[] {1}, 1, -1));
+    assertThat(nonEmptyBloomFilterException).hasMessageThat().contains("Invalid hash count: -1");
+  }
+
+  @Test
+  public void constructorShouldThrowIAEOnOverflowPadding() {
```

Comment:
nit: Suggest renaming `constructorShouldThrowIAEOnOverflowPadding()` to `constructorShouldThrowIAEIfPaddingIsTooLarge()`. To me, the term "overflow" (incorrectly) implies that the value was once valid but grew to size that overflowed the valid range.